src/components/ReanimatedDrawerLayout.tsx

```diff
@@ -0,0 +1,745 @@
+    // See the diagram for reference. * = starting finger position, < = current finger position
+    // 1) +---------------+ 2) +---------------+ 3) +---------------+ 4) +---------------+
+    //    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXX|.........|
+    //    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXX|.........|
+    //    |XXXXXXXX|..<*..|    |XXXXXXXX|.<-*..|    |XXXXXXXX|<--*..|    |XXXXX|<-----*..|
+    //    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXX|.........|
+    //    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXXXXX|......|    |XXXXX|.........|
+    //    +---------------+    +---------------+    +---------------+    +---------------+
+
+    const openValue = useDerivedValue(() => {
+      return interpolate(
+        drawerTranslation.value,
+        [0, drawerWidth],
+        [0, 1],
+        Extrapolation.CLAMP
+      );
+    });
+
+    const gestureOptions: {
+      useNativeDriver: boolean;
+      listener?: (event: { nativeEvent: { translationX: number } }) => void;
+    } = React.useMemo(
+      () => ({
+        useNativeDriver: useNativeAnimations,
+        listener: props.onDrawerSlide
+          ? (ev) => {
+              const translationX = Math.floor(
+                Math.abs(ev.nativeEvent.translationX)
+              );
+              const position = translationX / containerWidth;
+
+              props.onDrawerSlide?.(position);
+            }
+          : undefined,
+      }),
+      [containerWidth, props, useNativeAnimations]
+    );
+
+    if (props.onDrawerSlide) {
+      gestureOptions.listener = (ev) => {
+        const translationX = Math.floor(Math.abs(ev.nativeEvent.translationX));
+        const position = translationX / containerWidth;
+
+        props.onDrawerSlide?.(position);
+      };
+    }
+
+    const isDrawerOpen = useSharedValue(false);
+
+    const handleContainerLayout = ({ nativeEvent }: LayoutChangeEvent) => {
+      setContainerWidth(nativeEvent.layout.width);
+    };
+
+    const emitStateChanged = React.useCallback(
+      (newState: DrawerState, drawerWillShow: boolean) => {
+        'worklet';
+        props.onDrawerStateChanged &&
+          runOnJS(props.onDrawerStateChanged)?.(newState, drawerWillShow);
+      },
+      [props.onDrawerStateChanged]
+    );
+
+    const isDrawerShowing = useSharedValue(false);
+
+    const drawerAnimatedProps = useAnimatedProps(() => ({
+      accessibilityViewIsModal: isDrawerShowing.value,
+    }));
+
+    const overlayAnimatedProps = useAnimatedProps(() => ({
+      pointerEvents: isDrawerShowing.value
+        ? ('auto' as const)
+        : ('none' as const),
+    }));
+
+    // While the drawer is hidden, it's hitSlop overflows onto the main view by edgeWidth
+    // This way it can be swiped open even when it's hidden
+    const [edgeHitSlop, setEdgeHitSlop] = React.useState<HitSlop>(
+      isFromLeft ? { right: edgeWidth } : { left: edgeWidth }
+    );
```

Comment:
Can't this be handled by an animated style dependant on `isDrawerOpen`?

Reply:
Not sure what you mean by that, it's only used for configuring the `Pan` gesture, so it's preferable for it to be `State`, and it's only set in one place besides the initialiser, so I'm not sure if it's worth creating an animated style specifically for it if we can just use it's setter.

Could you please elaborate?

Reply:
I'll be honest, I don't remember what I meant 😅. But looking at it now, I don't see where the animated style would be used so I think we can safely ignore that one.